### DISCUSSION TITLE
【自动更新】希望添加自动更新github仓库的功能

### DISCUSSION THREAD

**Discussion by lveMonsi** [tangly1024/NotionNext#2087] [Ideas]
**为什么提出这个新的特性改动**
更新较为繁琐，若添加workflows自动更新仓库代码会更加方便

**描述一下你推荐的解决方案**
参考https://github.com/Yidadaa/ChatGPT-Next-Web/blob/main/.github/workflows/sync.yml
使用workflows定时更新仓库代码，博客配置则使用环境变量的方式

**Comment by Angel0726** [comment]
为什么自动更新，我现在更新notionnext仓库一点也不复杂，点一下同步就好。

**Comment by tangly1024** [comment]
手动更新不麻烦。然而这个能自动更新的方案很神奇，有没有懂得这个功能的大佬给我科普一下😁

**Comment by LooseLi** [comment]
> 自动更新不麻烦。然而这个能自动更新的方案很神奇，有没有懂得这个功能的大佬给我科普一下😁

我理解的自动更新仓库代码其实就是利用GitHub Action去定时执行将上游仓库代码同步至fork仓库这一任务

**Comment by nowscott** [comment]
> 手动更新不麻烦。然而这个能自动更新的方案很神奇，有没有懂得这个功能的大佬给我科普一下😁

大概看了一下，定时每天上午8点运行这个workflow，workflow的内容就是去被fork的库获取和当前库的区别，然后覆盖。其中需要自己设置一个仓库的secret，包含一个自己账户的token。
